src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -978,7 +978,19 @@ void MacroAssembler::li(Register Rd, int64_t imm) {
   }
 }
 
+void MacroAssembler::load_link(const address source, Register temp) {
+  assert(temp != noreg && temp != x0, "expecting a register");
+  assert(temp == x5, "expecting a register");
+  assert_cond(source != nullptr);
+  int64_t distance = source - pc();
+  assert(is_simm32(distance), "Must be");
```

Comment:
seems load_link can jump to about +/-2G dest from pc, jump_link seems support full address range jump (e.g. 48 bits)?

Reply:
The load can only happen within -+2Gb but address loaded is any 64 bit.